File: src/main/java/org/jukeboxmc/event/player/PlayerToggleSneakEvent.java
package org.jukeboxmc.event.player;

import org.jukeboxmc.event.Cancellable;
import org.jukeboxmc.player.Player;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class PlayerToggleSneakEvent extends PlayerEvent implements Cancellable {

    private final boolean isSneaking;

    /**
     * Creates a new {@link PlayerToggleSneakEvent}
     *
     * @param player     who toggled sneaking
     * @param isSneaking whether the player is sneaking
     */
    public PlayerToggleSneakEvent( Player player, boolean isSneaking ) {
        super( player );

        this.isSneaking = isSneaking;
    }

    /**
     * Retrieves whether the player is sneaking or not
     *
     * @return whether the player is sneaking
     */
    public boolean isSneaking() {
        return this.isSneaking;
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/event/player/PlayerToggleSneakEvent.kt
package org.jukeboxmc.api.event.player

import org.jukeboxmc.api.event.Cancellable
import org.jukeboxmc.api.player.Player

class PlayerToggleSneakEvent(
    player: Player,
    private val sneaking: Boolean
) : PlayerEvent(player), Cancellable {

    fun isSneaking(): Boolean {
        return this.sneaking
    }

}